bin/wasm-node/javascript/src/worker/bindings-smoldot-light.ts

```diff
@@ -0,0 +1,310 @@
+                        throw new ConnectionError('TCP connections not available');
+                    }
+
+                    const socket = compat.createConnection({
+                        host: tcpParsed[2],
+                        port: parseInt(tcpParsed[3] as string, 10),
+                    });
+
+                    connection = { ty: 'tcp', socket };
+                    connection.socket.setNoDelay();
+
+                    connection.socket.on('connect', () => {
+                        if (socket.destroyed) return;
+                        instance.exports.connection_open(id);
+                    });
+                    connection.socket.on('close', (hasError) => {
+                        if (socket.destroyed) return;
+                        // NodeJS doesn't provide a reason why the closing happened, but only
+                        // whether it was caused by an error.
+                        const message = hasError ? "Error" : "Closed gracefully";
+                        const len = Buffer.byteLength(message, 'utf8');
+                        const ptr = instance.exports.alloc(len) >>> 0;
+                        Buffer.from(instance.exports.memory.buffer).write(message, ptr);
+                        instance.exports.connection_closed(id, ptr, len);
+                    });
+                    connection.socket.on('error', () => { });
+                    connection.socket.on('data', (message) => {
+                        if (socket.destroyed) return;
+                        const ptr = instance.exports.alloc(message.length) >>> 0;
+                        message.copy(Buffer.from(instance.exports.memory.buffer), ptr);
+                        instance.exports.connection_message(id, ptr, message.length);
+                    });
+
+                } else {
+                    throw new ConnectionError('Unrecognized multiaddr format');
+                }
+
+                connections[id] = connection;
+                return 0;
+
+            } catch (error) {
+                const isBadAddress = error instanceof ConnectionError;
+                let errorStr = "Unknown error";
+                if (error instanceof Error) {
+                    errorStr = error.toString();
+                }
+                const mem = Buffer.from(instance.exports.memory.buffer);
+                const len = Buffer.byteLength(errorStr, 'utf8');
+                const ptr = instance.exports.alloc(len) >>> 0;
+                mem.write(errorStr, ptr);
+                mem.writeUInt32LE(ptr, error_ptr_ptr);
+                mem.writeUInt32LE(len, error_ptr_ptr + 4);
+                mem.writeUInt8(isBadAddress ? 1 : 0, error_ptr_ptr + 8);
+                return 1;
+            }
+        },
+
+        // Must close and destroy the connection object.
+        connection_close: (id: number) => {
+            let connection = connections[id] as TcpWrapped | WebSocketWrapped;
```

Comment:
the `as TcpWrapped | WebSocketWrapped` is not needed (and a bit misleading b/c someone could think that what's in `connections` is not that type), so I would remove it. If you wanted to be explicit about the type of `connection` you could do:

```
let connection: TcpWrapped | WebSocketWrapped = connections[id]
```

which is unnecessary, but at least is not misleading :slightly_smiling_face:

Reply:
Ups! I was wrong, when I wrote that ^^ I didn't realize that you were using the `noUncheckedIndexedAccess` option, sorry about that. Well, then I would go with: `connections[id]!`.